packages/firestore/src/api/database.ts

```diff
@@ -2442,7 +2442,7 @@ export class CollectionReference<T = firestore.DocumentData> extends Query<T>
     // We allow omission of 'pathString' but explicitly prohibit passing in both
     // 'undefined' and 'null'.
     if (arguments.length === 0) {
-      pathString = AutoId.newId();
+      pathString = AutoId.newId(PlatformSupport.getPlatform());
```

Comment:
`AutoId.newId` should just call `PlatformSupport.getPlatform()` for itself. If you need a version of this that takes the platform as an argument for testing, it's fine to have `AutoId.newId()` delegate to that version. We shouldn't have to update all callers to fix this.

Reply:
Done.